iroha_http_server/src/lib.rs

```diff
@@ -0,0 +1,640 @@
+                                                }
+                                            }
+                                        } else {
+                                            Some(HttpResponse::method_not_allowed(&[GET_METHOD]))
+                                        }
+                                    }
+                                    POST_METHOD => {
+                                        if let Endpoint::Post(handler) = endpoint {
+                                            match handler.call(state, params, request.clone()).await
+                                            {
+                                                Ok(response) => Some(response),
+                                                Err(err) => {
+                                                    log::error!("Failed to handle post request {:?} with error: {}", request, err);
+                                                    Some(HttpResponse::internal_server_error())
+                                                }
+                                            }
+                                        } else {
+                                            Some(HttpResponse::method_not_allowed(&[POST_METHOD]))
+                                        }
+                                    }
+                                    _ => Some(HttpResponse::method_not_allowed(&[
+                                        GET_METHOD,
+                                        POST_METHOD,
+                                    ])),
+                                }
+                            }
+                        } else {
+                            // Consume peeked data.
+                            consume_bytes(&mut stream, read_size as u64).await;
+                            Some(HttpResponse::not_found())
+                        }
+                    }
+                    Err(err) => {
+                        log::error!("Failed to parse incoming HTTP request: {}", err);
+                        //TODO: return `not supported` for the features that are not supported instead of bad reauest.
+                        Some(HttpResponse::bad_request())
+                    }
+                };
+                if let Some(response) = response {
+                    let bytes: Vec<u8> = response.borrow().into();
+                    if let Err(err) = stream.write_all(&bytes).await {
+                        log::error!("Failed to write back HTTP response: {}", err);
+                    }
+                }
+                // TODO: return the response over the stream if it is there
+            });
+        }
+        Ok(())
+    }
+}
+
+pub type WebSocketStream = TungsteniteWebSocketStream<TcpStream>;
+
+pub enum Endpoint<State> {
+    WebSocket(Box<dyn WebSocketHandler<State>>),
+    Get(Box<dyn HttpHandler<State>>),
+    Post(Box<dyn HttpHandler<State>>),
```

Comment:
for REST we will also need PUT, PATCH, DELETE, etc. - may be just depicted as TODO for now

Reply:
Or even better to make it another level
Endpoint:
- WebSocket
- HTTP
--  Get
-- ...